interactions/api/models/webhook.py

```diff
@@ -0,0 +1,296 @@
+        :type username: str
+        :param avatar_url: override the default avatar of the webhook
+        :type avatar_url: str
+        :param tts: true if this is a TTS message
+        :type tts: bool
+        :param embeds: embedded ``rich`` content
+        :type embeds: Union[Embed, List[Embed]]
+        :param allowed_mentions: allowed mentions for the message
+        :type allowed_mentions: dict
+        :param components: the components to include with the message
+        :type components: Union[ActionRow, Button, SelectMenu, List[ActionRow], List[Button], List[SelectMenu]]
+        :param files: The files to attach to the message
+        :type files: Union[File, List[File]]
+        :param thread_id: Send a message to a specified thread within a webhook's channel. The thread will automatically be unarchived
+        :type thread_id: int
+        """
+
+        if not self._client:
+            raise AttributeError("HTTPClient not found!")
+
+        from ...client.models.component import _build_components
+        from .message import Message
+
+        _content: str = "" if content is MISSING else content
+        _tts: bool = False if tts is MISSING else tts
+        # _attachments = [] if attachments else None
+        _embeds: list = (
+            []
+            if not embeds or embeds is MISSING
+            else ([embed._json for embed in embeds] if isinstance(embeds, list) else [embeds._json])
+        )
+        _allowed_mentions: dict = {} if allowed_mentions is MISSING else allowed_mentions
+
+        if not components or components is MISSING:
+            _components = []
+        else:
+            _components = _build_components(components=components)
+
+        if not files or files is MISSING:
+            _files = []
+        elif isinstance(files, list):
+            _files = [file._json_payload(id) for id, file in enumerate(files)]
+        else:
+            _files = [files._json_payload(0)]
+            files = [files]
+
+        msg = Message(
+            content=_content,
+            tts=_tts,
+            attachments=_files,
+            embeds=_embeds,
+            components=_components,
+            allowed_mentions=_allowed_mentions,
+        )
+
+        payload = msg._json
+
+        if username is not MISSING:
+            payload["username"] = username
+
+        if avatar_url is not MISSING:
+            payload["avatar_url"] = avatar_url
+
+        res = await self._client.execute_webhook(
+            webhook_id=int(self.id),
+            webhook_token=self.token,
+            files=files,
+            payload=payload,
+            thread_id=thread_id if thread_id is not MISSING else None,
+        )
+
+        return Message(**res, _client=self._client)
+
+    async def delete(self) -> None:
+        """
+        Deletes the webhook.
+        """
+
+        await self._client.delete_webhook(webhook_id=int(self.id), webhook_token=self.token)
```

Comment:
Add a check to see if self._client exists
```suggestion
    async def delete(self) -> None:
        """
        Deletes the webhook.
        """
        if not self._client:
            raise AttributeError("HTTPClient not found!")

        await self._client.delete_webhook(webhook_id=int(self.id), webhook_token=self.token)
```